content/blog/0012-effect-3.0.mdx

```diff
@@ -0,0 +1,328 @@
+---
+title: Effect 3.0 (Release)
+excerpt: Production-Grade Software in TypeScript with Effect
+date: 2024-03-16
+authors:
+  - name: Michael Arnaldi
+    description: Chief Executive Officer
+    avatar: https://pbs.twimg.com/profile_images/1232338149340319751/Ln9nNWyk_400x400.jpg
+    twitter: https://twitter.com/MichaelArnaldi
+---
+
+## Announcement
+
+Effect is finally Production Ready!
+
+I am pleased to announce that after 5 years of work we are ready to release Effect 3.0, the first stable release of the core of the Effect Ecosystem.
+
+Starting with 3.0 the main package `effect` will follow semantic versioning:
+
+1) major releases will include breaking changes
+2) minor releases will include new features and new modules
+3) patch releases will include bug-fixes
+
+We do not expect new major release in the near future, we will only release a new major when a significant number of API improvements have been made and after a long period of feedback.
+
+## Quickstart
+
+Effect is a TypeScript library that works in every runtime & project, to start using it you can:
+
+```bash
+npm install effect
+```
+
+And follow our [docs](https://effect.website/docs/quickstart).
+
+If you prefer instead to have a look at a fully working & effect-native app we've prepared a [demo cli app](https://github.com/IMax153/effect-openai) that you can directly open in [Gitpod](https://gitpod.io/) or locally (if you prefer), you'll need to provide an OpenAI API Key in order to integrate with the OpenAI API. The demo app allows you to train a model via embeddings from a set of files and then allows you to prompt the trained model with questions.
```

Comment:
For what it's worth, I think this is the perfect example / demo. But we can iterate on this a bit to make it more accessible still (it should feature a "tour guide" documentation that shows you around and invites you to experiment with it). It would also be cool to further expand it with some custom metrics (not just span metrics & traces) and give the pre-configured grafana dashboard a bit more love.

The "tour guide" is imho the most important bit though so it's not just our usual "here's some code, go check it out by yourself".